### PR TITLE
Multiple fixes for UCX v1.5.2

### PR DESCRIPTION
**Pull request by yosefe**
Fixing multiple issues for redhat per https://bugzilla.redhat.com/show_bug.cgi?id=1715129
- Static checker: memory pool allocation falures, redundant code, ucx_info alias print
- Fixing segfault reported by Fedora when OpenMPI is compiled with UCX

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10492/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10501/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
@brminich can you pls take a look?

File: test/apps/test_dlopen_cfg_print.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    /* unload and reload uct while ucs is loaded
+     * would fail if uct global vars are kept on global lists in ucs */
+    ucs_handle = do_dlopen_or_exit(ucs_filename);
+    for (i = 0; i < 2; ++i) {
+        uct_handle = do_dlopen_or_exit(uct_filename);
+        dlclose(uct_handle);
+    }
+
+    /* print all config table, to force going over the global list in ucs */
+    void (*print_all_opts)(FILE*,int) = dlsym(ucs_handle,
```
Comment (brminich) [inline test/apps/test_dlopen_cfg_print.c]: minor: space after `,`

**Pull request merged**